libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -339,11 +339,12 @@ WaveTrack::IntervalHolder WaveTrack::Interval::GetStretchRenderedCopy(
    dst->ClearRight(originalPlayEndTime);
 
    // We don't preserve cutlines but the relevant part of the envelope.
-   Envelope dstEnvelope = dst->GetEnvelope();
+   Envelope dstEnvelope = GetEnvelope();
    const auto samplePeriod = 1. / mpClip->GetRate();
    dstEnvelope.CollapseRegion(0, originalPlayStartTime, samplePeriod);
    dstEnvelope.CollapseRegion(
       originalPlayEndTime, GetSequenceEndTime() + samplePeriod, samplePeriod);
```

Comment:
I think you may need to swap the two lines above to get correct results in all cases.  Because the function move envelope points after the collapsed part leftward.

Reply:
I wouldn't expect that, but to be on the safe side I'll push that change.